File: WoofAndroid/data/repositories/src/main/kotlin/com/perrystreet/woof/repositories/dogs/mapper/DogsPageDTOToDomainMapper.kt
package com.perrystreet.woof.repositories.dogs.mapper

import com.perrystreet.woof.dto.dog.DogsPageDTO
import com.perrystreet.woof.models.dog.DogsPage
import org.koin.core.annotation.Factory

@Factory
class DogsPageDTOToDomainMapper(
    private val dogMapper: DogDTOToDomainMapper,
) {
    operator fun invoke(dto: DogsPageDTO): DogsPage = DogsPage(
        dogs = dto.results.map { dogMapper(it) },
        offset = dto.offset,
        total = dto.total,
    )
}

File: WoofIOS/SwiftPackages/Data/Repositories/Sources/Repositories/Dogs/Mapper/DogsPageDTOToDomainMapper.swift
import DI
import DTO
import Models

@Factory
public final class DogsPageDTOToDomainMapper {
    private let dogMapper: DogDTOToDomainMapper

    public func callAsFunction(_ dto: DogsPageDTO) -> DogsPage {
        DogsPage(
            dogs: dto.results.map { dogMapper($0) },
            offset: dto.offset,
            total: dto.total
        )
    }
}
